keyring/backends/SecretService.py

```diff
@@ -33,11 +34,13 @@ def priority(cls):
         try:
             bus = secretstorage.dbus_init()
             list(secretstorage.get_all_collections(bus))
+            bus.sock.close()
```

Comment:
I feel uneasy about this change. It's not obvious to me when `bus.sock` gets opened... or when it should be closed. Does it only need to be closed when iterating on `get_all_collections` succeeds? That's what this does.